content/docs/5.1/concepts/explore-tikv-features/cas.md

```diff
@@ -0,0 +1,137 @@
+```
+
+2. If TiUP is already installed, update the TiUP playground component to the latest version:
+
+```bash
+tiup update --self && tiup update playground
+```
+
+3. Install Rust
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+
+## Step 1: Start TiKV Cluster
+
+For the purpose of this tutorial, you need only one TiKV node, so use the `tiup playground` command.
+
+Show TiUP version:
+
+```bash
+tiup -v
+```
+
+version >= 1.5.0:
+
+```bash
+tiup playground --mode tikv-slim
+```
+
+version < 1.5.0:
+
+```bash
+tiup playground
+```
+
+## Step 2: Create a rust project
+
+Use `cargo` to create a rust project:
+
+```bash
+cargo new hello-tikv
+cd hello-tikv
+```
+
+## Step 3: Add dependency
+
+Append the following content to file `Cargo.toml`.
+
+```toml
+tikv-client = "0.1.0"
+tokio = { version = "1.5.0", features = ["full"] }
+```
+
+## Step 4: Write test code
+
+Save the following code to file `src/main.rs`.
+
+```rust
+use tikv_client::RawClient;
+
+#[tokio::main]
+async fn main() {
+    let client = RawClient::new(vec!["127.0.0.1:2379"])
+        .await.unwrap()
+        // Users must set `with_atomic_for_cas` when using `put`, `delete`,
+        // `batch_delete` and `batch_put` to ensure linearizability
+        // of `compare_and_swap`.
+        .with_atomic_for_cas();
+
+    let key = "key".to_owned();
+    let value = "value".to_owned();
+    let new_value = "newValue".to_owned();
+
+    // call put(key, value)
+    client.put(key.to_owned(), value.to_owned()).await.unwrap();
+    println!("put key={}, value={}", key, value);
+
+    // call get(key) returns value
+    let res = client.get(key.to_owned()).await.unwrap();
+    println!("get key={} returnedValue {:?}", key, res);
+
+    // call compare_and_swap(key, value, new_value) returns (value, true)
+    // If the value retrived is equal to `current_value`, `new_value` is written.
+    // A tuple is returned if successful: the previous value and whether the value is swapped
+    let res = client
+        .compare_and_swap(
+            key.to_owned(),
+            Some(value.to_owned()).map(|v| v.into()),
+            new_value.to_owned(),
+        )
+        .await.unwrap();
+    println!("compare_and_swap key={} oldValue={} newValue={} returnedValue={:?}", key, value, new_value, res);
+
+    // call get(key) returns new_value
+    let res = client.get(key.to_owned()).await.unwrap();
+    println!("get key={} returnedValue={:?}", key, res);
+}
+```
+
+## Step 5: Run the code
+
+```bash
+cargo run
+
+put key=key, value=value
+get key=key returnedValue Some([118, 97, 108, 117, 101])
+compare_and_swap key=key oldValue=value newValue=newValue returnedValue=(Some([118, 97, 108, 117, 101]), true)
+get key=key returnedValue=Some([110, 101, 119, 86, 97, 108, 117, 101])
+```
+
+As we can see, after calling `compare_and_swap` the `value` is replaced by `newValue`.
+
+{{< warning >}}
```

Comment:
Seems this warning is already in example code comments? Could we remove this and add a **NOTE** header in the comments?

Reply:
It's a bit repetitive. I remove the comments in code.